invenio_drafts_resources/records/systemfields/versions.py

```diff
@@ -0,0 +1,259 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) 2021 CERN.
+#
+# Invenio-Drafts-Resources is free software; you can redistribute it and/or
+# modify it under the terms of the MIT License; see LICENSE file for more
+# details.
+
+"""Versions field."""
+
+import uuid
+from copy import copy
+
+from invenio_db import db
+from invenio_records.systemfields import RelatedModelField
+from invenio_records.systemfields.base import SystemField
+from sqlalchemy import inspect
+from sqlalchemy.exc import IntegrityError
+
+
+def uuid_or_none(val):
+    """Convert string to UUID object."""
+    if val is not None and not isinstance(val, uuid.UUID):
+        return uuid.UUID(val)
+    return val
+
+
+class VersionsManager:
```

Comment:
Comment: Later we could add more methods in order to query for the other records via the database etc.